pyramid_openapi3/__init__.py

```diff
@@ -330,35 +333,47 @@ def check_all_routes(event: ApplicationCreated):
             logger.info("Endpoint validation against specification is disabled")
             return
 
-        # Sometimes api routes are prefixed with `/api/v1` and similar, using
-        # https://swagger.io/docs/specification/api-host-and-base-path/
-        prefixes = []
-        for server in openapi_settings["spec"].servers:
-            path = urlparse(server.url).path
-            if path != "/":
-                prefixes.append(path)
+        prefixes = _get_server_prefixes(openapi_settings["spec"])
 
         def remove_prefixes(path):
             path = f"/{path}" if not path.startswith("/") else path
             for prefix in prefixes:
-                path = path.replace(prefix, "")
+                if path.startswith(prefix):
+                    prefix_length = len(prefix)
+                    return path[prefix_length:]
             return path
 
         paths = list(openapi_settings["spec"].paths.keys())
         routes = [
-            remove_prefixes(route.path)
-            for name, route in app.routes_mapper.routes.items()
+            remove_prefixes(route.path) for route in app.routes_mapper.routes.values()
         ]
-        route_names = {
-            remove_prefixes(route.path): name
-            for name, route in app.routes_mapper.routes.items()
-        }
 
         missing = [r for r in paths if r not in routes]
         if missing:
             raise MissingEndpointsError(missing)
 
         settings.setdefault("pyramid_openapi3", {})
         settings["pyramid_openapi3"].setdefault("routes", {})
-        for path in paths:
-            settings["pyramid_openapi3"]["routes"][route_names[path]] = name
+
+        # It is possible to have multiple `add_route` for a single path
+        # (due to request_method predicates). So loop through each route
+        # to create a lookup of route_name -> api_name
+        for route_name, route in app.routes_mapper.routes.items():
+            if remove_prefixes(route.path) in paths:
+                settings["pyramid_openapi3"]["routes"][route_name] = name
+
+
+def _get_server_prefixes(spec: Spec) -> t.List[str]:
+    """
+    Build a set of possible route prefixes from the api spec.
```

Comment:
```suggestion
    """Build a set of possible route prefixes from the api spec.
```

Strange that linters didn't complain about this one. Usually we start the docstring without a new line.

Reply:
I seem to only get the error if it is a single line docstring (D200). I am not getting the H404 error, strange.